lib/jobs/dell-wsman-add-hotspare-updatexml.js

```diff
@@ -0,0 +1,162 @@
+// Copyright © 2017 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+'use strict';
+
+var di = require('di'),
+    xmldom = require('xmldom').DOMParser;
+
+module.exports = WsmanAddHotspareUpdateXmlFactory;
+
+di.annotate(WsmanAddHotspareUpdateXmlFactory, new di.Provide('Job.Dell.Wsman.Add.Hotspare.UpdateXml'));
+di.annotate(WsmanAddHotspareUpdateXmlFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Assert',
+    'Util',
+    'Services.Configuration',
+    'Errors',
+    'JobUtils.Smb2Client',
+    'JobUtils.NfsClient',
+    'Promise',
+    '_'
+));
+
+function WsmanAddHotspareUpdateXmlFactory(
+    BaseJob,
+    Logger,
+    assert,
+    util,
+    configuration,
+    errors,
+    Smb2Client,
+    NfsClient,
+    Promise,
+    _
+) {
+    var logger = Logger.initialize(WsmanAddHotspareUpdateXmlFactory);
+
+    function WsmanAddHotspareUpdateXmlJob(options, context, taskId) {
+        WsmanAddHotspareUpdateXmlJob.super_.call(this, logger, options, context, taskId);
+    }
+
+    util.inherits(WsmanAddHotspareUpdateXmlJob, BaseJob);
+
+    WsmanAddHotspareUpdateXmlJob.prototype._run = function() {
+        var self = this;
+        if(!self.options.driveId) {
+            throw new errors.NotFoundError('The driveId should not be empty.');
+        }
+        self.dell = configuration.get('dell');
+        if(!self.dell || !self.dell.shareFolder) {
+            throw new errors.NotFoundError('Share folder is not defined in smiConfig.json');
+        }
+        return self.updateXmlForRaid();
+    };
+
+    WsmanAddHotspareUpdateXmlJob.prototype.updateXmlForRaid = function() {
+        var self = this;
+        var fileName = self.context.graphId + '.xml';
+        var client;
+        return Promise.try(function() {
+            if(self.dell.shareFolder.shareType === 0) {
+                client = new NfsClient(
+                    self.dell.shareFolder.address,
+                    self.dell.shareFolder.shareName,
+                    self.context.mountDir
+                );
+            } else if(self.dell.shareFolder.shareType === 2) {
```

Comment:
shareType `0` and `2` is better to defined as a constants

Reply:
Is it the common constants in on-core repo?
https://github.com/RackHD/on-core/blob/master/lib/common/constants.js